torch/export/_trace.py

```diff
@@ -105,6 +105,29 @@ def _ignore_backend_decomps():
         torch.backends.nnpack.set_flags(*orig_nnpack_flag)
 
 
+@contextmanager
+def _patch_dropout_decomps():
+    """
+    In export, we always want to decompose dropout_
+    to functional dropout. The very correct way to do is
+    we extend functionalization to automatically functionalize
+    CompositeImplicitAutograd ops using their functional variant.
+    But this is significantly lot more work to do. Since, it seems
+    users only care about certain set of ops, so for now we just
+    patch our own CompositeImplicitAutograd decomp
+    """
+    def dropout_(input, p, train):
+        res = torch.ops.aten.dropout.default(input, p, train)
+        input.copy_(res)
+        return res
+
+    torch.ops.aten.dropout_.default.py_impl(torch._C.DispatchKey.CompositeImplicitAutograd)(dropout_)
```

Comment:
cc @ezyang is this the right way to temporarily patch decomp?

Reply:
No, this does not look like it will invalidate the cache so you will not successfully unset the decomp when you are done